modules/standard/Buffers.chpl

```diff
@@ -140,18 +141,35 @@ module Buffers {
      :arg error: (optional) capture an error that was encountered instead of
                  halting on error
    */
-  proc bytes.bytes(len:int(64), out error:syserr) {
+  proc bytes.init(len:int(64), out error:syserr) {
+    this.home = here;
+    super.init();
     error = qbytes_create_calloc(this._bytes_internal, len);
     // The buffer is "retained" internally on creation, but only on success.
-    this.home = here;
   }
   pragma "no doc"
-  proc bytes.bytes(len:int(64)) {
-    var error:syserr = ENOERR;
-    error = qbytes_create_calloc(this._bytes_internal, len);
+  proc bytes.init(len:int(64)) {
+    this.home = here;
+    super.init();
+    var error:syserr = qbytes_create_calloc(this._bytes_internal, len);
     if error then try! ioerror(error, "in bytes constructor");
     // The buffer is retained internally on construction, but only on success.
-    this.home = here;
+  }
+
+
+  pragma "no doc"
+  proc bytes.init(x: bytes) {
+    if x.home == here {
+      this.home = here;
+      this._bytes_internal = x._bytes_internal;
+      super.init();
+      qbytes_release(x._bytes_internal);
+    } else {
+      this.home = here;
```

Comment:
If we always set `this.home` to `here`, why not do that outside the if statement?

Reply:
I guess because I was trying to retain as much of the original code as possible (not that it helped minimize the diffs) and didn't really expend any effort to try and understand what it did.